transcom/mymove
# [ MB-16998 ] Data model updates for pending billed start date

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16998)

## Summary

- [x] Adding the PendingBilledStartDate to the API;
  - This patch makes a number of changes to the GHC API around adding pending billed
    SIT start date into the response for shipment payments SIT balance.
- [x] Adding pending billed start date to the service

This patch series adds the pending billed start date to the response for
shipment payments SIT balances endpoint. This work can be tested by an engineer
locally or in an ephemeral deployment, but only in what gets returned in the
endpoint call for the `/shipments-payment-sit-balance`.

➡️ [shipments-payment-sit-balance](https://transcom.github.io/mymove-docs/api/ghc#tag/paymentRequests/operation/getShipmentsPaymentSITBalance)

```txt
/v1/payment-requests/${PAYMENT_REQUEST_ID}/shipments-payment-sit-balance
```

When verifying the response comes back from the server appropriately, open the
**Network** tab in the _Chrome Debugger_ tool and load a payment request page as
a _TIO_ with a similar URL like, `/moves/${LOCATOR}/payment-requests/${UUID}`.

If you get stuck reviewing this, please reach out to me directly to walk you
through it.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Create a multi-user, if you don't have one locally.
1. Navigate to the Testharness data scenarios page.
1. Create a new Move using the _HHGMoveInSIT_ button.
1. As the _TOO_,review and accept the Domestic Destination SIT Delivery address
   update request.
1. As the _TOO_, review the SIT start date as you will be using this as the date
   window we'll need for the _Prime_ Payment Request.
1. As the _Prime_, navigate to the Move and create a Payment Request using the
   button at the top.
1. As the _Prime_, use the **SIT start date** mentioned above to select a date
   for the DDASIT payment start date a day later than the date recorded in the
   above step. Set a date for payment end date _after the date_ you set for
   payment start date.
1. As the _TIO_, navigate to the move using the locator to see the payment
   request. When you open the document viewer, see the **Network** tab in the
   _Chrome Debugger_ tools and search for the endpoint named
   `shipments-payment-sit-balance`. Notice that the server response now includes
   a key of `pendingBilledStartDate` with the value you entered as the _Prime_
   for the payment start date.
